### PR TITLE
BUG: fix to_records confict with unicode_literals #13172

### LINKED ISSUE #13172: COMPAT: unicode_literals conflict with to_records
#### Code Sample, a copy-pastable example if possible

```
from __future__ import unicode_literals
import pandas as pd
pd.DataFrame([{'id': '10', 'title': 'hello'}]).set_index('id').to_records()
```
the code above will report following error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-40-056b33fad34d> in <module>()
----> 1 pd.DataFrame([{'id': '10', 'title': 'hello'}]).set_index('id').to_records()

/usr/local/lib/python2.7/site-packages/pandas/core/frame.pyc in to_records(self, index, convert_datetime64)
   1068             names = lmap(str, self.columns)
   1069
-> 1070         dtype = np.dtype([(x, v.dtype) for x, v in zip(names, arrays)])
   1071         return np.rec.fromarrays(arrays, dtype=dtype, names=names)
   1072

TypeError: data type not understood
```

If I comment out `from __future__ import unicode_literals`, the code above will work fine.

```
rec.array([('10', 'hello')],
          dtype=[('id', 'O'), ('title', 'O')])
```

#### output of ``pd.show_versions()``
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Darwin
OS-release: 15.4.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: zh_CN.UTF-8

pandas: 0.18.1
nose: None
pip: 8.1.1
setuptools: 19.4
Cython: None
numpy: 1.11.0
scipy: None
statsmodels: None
xarray: None
IPython: 4.2.0
sphinx: 1.3.5
patsy: None
dateutil: 2.5.3
pytz: 2016.4
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.4.1
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None
pandas_datareader: None

### PR DESCRIPTION
**Pull request by starplanet**
- [ ] closes #13172 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
pls add the test from the issue & a whats new (bug fix)

File: pandas/tests/frame/test_convert_to.py [hunk 1]
```diff
@@ -172,3 +172,9 @@ def test_to_records_index_name(self):
         df.index.names = ['A', None]
         rs = df.to_records()
         self.assertIn('level_0', rs.dtype.fields)
+
+    def test_to_records_with_unicode_index(self):
+        res = DataFrame([{u'a': u'x', u'b': 'y'}]).set_index(u'a').to_records()
+        expect_res = np.rec.array([('x', 'y')], dtype=[('a', 'O'), ('b', 'O')])
+        self.assertEqual(res, expect_res)
```
Comment (jreback) [inline pandas/tests/frame/test_convert_to.py]: use 
```
result = ....
expected = ....
tm.assert_numpy_array_equal(result, expected)
```

use ``tm.assert_numpy_array_equal`` here (and only need 1 assert; that will compare everything)

add the issue number as a comment

**Comment by jreback** [comment]
lgtm. minor test change; ping when green.

**Comment by starplanet** [comment]
done.
原始邮件
发件人:Jeff Rebacknotifications@github.com
收件人:pydata/pandaspandas@noreply.github.com
抄送:starplanetzhangjinjie@yimian.com.cn; Authorauthor@noreply.github.com
发送时间:2016年5月16日(周一) 20:15
主题:Re: [pydata/pandas] BUG: fix to_records confict with unicode_literals#13172 (#13178)


lgtm. minor test change; ping when green.

**Comment by jreback** [comment]
thanks @starplanet nice patch!

**Pull request closed without merging**